specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2022-06-30-preview/iotcentral.json

```diff
@@ -0,0 +1,8857 @@
+      "type": "object",
+      "properties": {
+        "tenantId": {
+          "description": "The AAD tenant ID of the AD Group.",
+          "type": "string"
+        },
+        "objectId": {
+          "description": "The AAD object ID of the AD Group.",
+          "type": "string"
+        }
+      },
+      "required": [
+        "tenantId",
+        "objectId"
+      ]
+    },
+    "AnyValue": {
+      "description": "Can be anything: string, number, array, object, etc. (except `null`)"
+    },
+    "ApiToken": {
+      "description": "The access token definition.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Permission"
+        }
+      ],
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "Unique ID of the API token.",
+          "type": "string",
+          "readOnly": true
+        },
+        "token": {
+          "description": "Value of the API token.",
+          "type": "string",
+          "readOnly": true
+        },
+        "expiry": {
+          "description": "String-formatted date representing the time when the token expires.",
+          "type": "string",
+          "format": "date-time"
+        }
+      }
+    },
+    "ApiTokenCollection": {
+      "description": "The paged results of API tokens.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The collection of API tokens.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ApiToken"
+          }
+        },
+        "nextLink": {
+          "description": "URL to get the next page of API tokens.",
+          "type": "string"
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "Attestation": {
+      "description": "The attestation definition.",
+      "type": "object",
+      "properties": {
+        "type": {
+          "description": "Type of the attestation.",
+          "type": "string"
+        }
+      },
+      "required": [
+        "type"
+      ],
+      "discriminator": "type"
+    },
+    "BarChartConfiguration": {
+      "description": "Configuration specifying options for a bar chart tile.",
+      "x-ms-discriminator-value": "barChart",
+      "allOf": [
+        {
+          "$ref": "#/definitions/TileConfiguration"
+        },
+        {
+          "$ref": "#/definitions/GroupTileConfiguration"
+        },
+        {
+          "$ref": "#/definitions/TileCapabilityConfiguration"
+        }
+      ],
+      "type": "object",
+      "properties": {
+        "format": {
+          "description": "The format configuration of the chart",
+          "$ref": "#/definitions/ChartFormatConfiguration"
+        },
+        "queryRange": {
+          "description": "The query range configuration of the chart",
+          "$ref": "#/definitions/TimeQueryRangeConfiguration"
+        }
+      },
+      "required": [
+        "queryRange"
+      ]
+    },
+    "BlobStorageV1Destination": {
```

Comment:
@yang-hai-feng -- qq for you on something i just noticed... Why does the reference to `BlobStorageV1Destination` have the `V1` in it? This seems like it's exposing an implementation detail that can be hidden from the developer. Similar question for `DataExplorerV1Destination`

Reply:
Hi @markweitzel, this is currently designed for versioning the same type of destination if there is break changes in future, for example if we had new DataExplorerV2Destination defined with required broken shape, it requires to work together with the existed DataExplorerV1Destination, all contracts are exposed to end user directly to ensure their client code continue to work on v1 destination and also be able to create the v2 one. Also, we are holding this part in preview as now we are working on re-purpose the data export to a more generic concept of rule engine platform, the new destination could be an email, icm, flow trigger etc. We are preparing to join another API review office hours for this part changes when it is ready. Besides, we also got comments from 1.1-preview that our naming of entity (Data export related) is too long in this part, will be reconsider at same time of rule engine alignment.